homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -69,25 +67,25 @@
 
     hass.data.setdefault(DOMAIN, {})
     accesspoints = config.get(DOMAIN, [])
+
     for conf in accesspoints:
+        apid = conf[CONF_ACCESSPOINT].replace('-', '').upper()
```

Comment:
We should not set up anything inside the setup method. Instead, if we spot authentication inside the setup method of a component, we should import it as a config entry. Look at how the Hue component does it.